app/models/embed/purl.rb

```diff
@@ -203,6 +203,16 @@ def files
         end
       end
 
+      # get first file of appropriate mime type
+      def media_file
+        files.find { |file| file.mimetype.match "^#{type}\/" }
```

Comment:
IIRC i was once told to prefer `find` over `detect`.  whatever people want.  rubocop didn't complain, FWIW.  does not matter to me at all.